hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncClientScanner.java

```diff
@@ -206,15 +246,34 @@ private void openScanner() {
     addListener(timelineConsistentRead(conn.getLocator(), tableName, scan, scan.getStartRow(),
       getLocateType(scan), this::openScanner, rpcTimeoutNs, getPrimaryTimeoutNs(), retryTimer,
       conn.getConnectionMetrics()), (resp, error) -> {
-        if (error != null) {
-          consumer.onError(error);
-          return;
+        final Span localSpan = span;
+        try (Scope ignored = localSpan.makeCurrent()) {
+          if (error != null) {
+            try {
+              consumer.onError(error);
+              return;
+            } finally {
+              TraceUtil.setError(localSpan, error);
+              localSpan.end();
+            }
+          }
+          startScan(resp);
         }
-        startScan(resp);
       });
   }
 
   public void start() {
-    openScanner();
+    final Span localSpan = new TableOperationSpanBuilder(conn)
```

Comment:
The only place where we create this scanner and call start on it is in RawAsyncTableImpl.scan method, where both these two operations are in the same thread, so I think here we could make the Span field final and create it in the constructor, so we do not need to care about concurrency access any more.

Reply:
Hmm yes, I suppose so.